sips/sip-02x/sip-02x-non-sequential-multisig-transactions.md

```diff
@@ -0,0 +1,643 @@
+
+This is intended to be an update and replacement for the existing "[Transaction Encoding](https://github.com/stacksgov/sips/blob/main/sips/sip-005/sip-005-blocks-and-transactions.md#transaction-encoding)" section of SIP-005.
+For anything not mentioned here, the rules from SIP-005 still apply.
+
+### Transaction Encoding
+
+A transaction includes the following information.  Multiple-byte fields are
+encoded as big-endian.
+
+* A 1-byte **version number**, identifying whether or not the transaction is
+  meant as a mainnet or testnet transaction.
+* A 4-byte **chain ID**, identifying which Stacks chain this transaction is
+  destined for.
+* A **transaction authorization** structure, described below, which encodes the
+  following information (details are given in a later section):
+   * The address of the origin account.
+   * The signature(s) and signature threshold for the origin account.
+   * The address of the sponsor account, if this is a sponsored transaction.
+   * The signature(s) and signature threshold for the sponsor account, if given.
+   * The **fee** to pay, denominated in microSTX.
+* A 1-byte **anchor mode**, identifying how the transaction should be mined.  It
+  takes one of the following values:
+   * `0x01`:  The transaction MUST be included in an anchored block
+   * `0x02`:  The transaction MUST be included in a microblock
+   * `0x03`:  The leader can choose where to include the transaction.
+* A 1-byte **post-condition mode**, identifying whether or not post-conditions
+  must fully cover all transferred assets.  It can take the following values:
+   * `0x01`:  This transaction may affect other assets not listed in the
+     post-conditions.
+   * `0x02`:  This transaction may NOT affect other assets besides those listed
+     in the post-conditions.
+* A length-prefixed list of **post-conditions**, describing properties that must be true of the
+  originating account's assets once the transaction finishes executing.  It is encoded as follows:
+   * A 4-byte length, indicating the number of post-conditions.
+   * A list of zero or more post-conditions, whose encoding is described below.
+* The **transaction payload**, described below.
+
+#### Version Number
+
+The version number identifies whether or not the transaction is a mainnet or
+testnet transaction.  A mainnet transaction MUST have its highest bit cleared, and a
+testnet transaction MUST have the highest bit set (i.e., `version & 0x80` must be
+non-zero for testnet, and zero for mainnet).  The lower 7 bits are ignored for
+now.
+
+#### Chain ID
+
+The chain ID identifies which instance of the Stacks blockchain this transaction
+is destined for.  Because the main Stacks blockchain and Stacks app chains
+(described in a future SIP) share the same transaction wire format, this field
+is used to distinguish between each chain's transactions.  Transactions for the
+main Stacks blockchain MUST have a chain ID of `0x00000000`.
+
+#### Transaction Authorization
```

Comment:
Hi @jbencin, you only need to include the description of the _new_ "Transaction Authorization" data this SIP presents.  You don't need anything else.  Specifically, the reader needs to be able to read this section, and know exactly how to construct byte-for-byte a valid, well-formed `TransactionAuth` that supports non-sequential signatures.

Reply:
@jcnelson I guess I misunderstood what you were asking for and included too much this time. I've updated it again so that it this SIP only contains the "Transaction Authorization" and "Transaction Signing and Verifying" sections. Because it was easy to do, these sections also contain details for the old multisig formats, so that it's no longer necessary for the user to read the corresponding sections of SIP-005